tests/library/test_json_order.py

```diff
@@ -0,0 +1,53 @@
+import json
+import os
+import unittest
+
+
+class TestJsonOrder(unittest.TestCase):
+
+    def test_json_key_order(self):
+        correct_order = [
+            "author",
+            "canonical_url",
+            "site_name",
+            "host",
+            "language",
+            "title",
+            "ingredients",
+            "ingredient_groups",
+            "instructions",
+            "instructions_list",
+            "category",
+            "yields",
+            "description",
+            "total_time",
+            "cook_time",
+            "prep_time",
+            "cuisine",
+            "cooking_method",
+            "ratings",
+            "ratings_count",
+            "equipment",
+            "reviews",
+            "nutrients",
+            "image",
+            "keywords",
+        ]
+        json_directory = "tests/test_data/"
+
+        for root, dirs, files in os.walk(json_directory):
+            for filename in files:
+                if filename.endswith(".json"):
+                    with open(os.path.join(root, filename), encoding="utf-8") as f:
+                        data = json.load(f)
+
+                    filtered_data_keys = [key for key in correct_order if key in data]
+
+                    expected_keys = filtered_data_keys
+                    actual_keys = list(data.keys())
+
+                    try:
+                        self.assertEqual(expected_keys, actual_keys)
+                    except AssertionError as e:
+                        e.args = (f"File: {os.path.join(root, filename)}\n" + str(e),)
+                        raise
```

Comment:
Sorta a nitpick here: I think it might be clearer to throw a newly-constructed `AssertionError` exception, but to maintain a reference to the original by using the `raise ... from` syntax:

https://docs.python.org/3/tutorial/errors.html#exception-chaining

The reason I think that is that adjusting the `args` on an existing exception object feels a little fragile to me.